trinity/rpc/ipc.py

```diff
@@ -88,44 +102,27 @@ def strip_non_json_prefix(raw_request):
     await writer.drain()
 
 
-def start(path, chain=None, loop=None):
-    '''
-    :returns: initialized server
-    :rtype: asyncio.Server
-    '''
-    if loop is None:
-        loop = asyncio.get_event_loop()
-    rpc = RPCServer(chain)
-    server = loop.run_until_complete(asyncio.start_unix_server(
-        connection_handler(rpc.execute),
-        path,
-        loop=loop,
-        limit=MAXIMUM_REQUEST_BYTES,
-    ))
-    return server
-
-
-def run_until_interrupt(server, loop=None):
-    logger = logging.getLogger('trinity.rpc.ipc')
-    if loop is None:
-        loop = asyncio.get_event_loop()
-    try:
-        loop.run_forever()
-    except KeyboardInterrupt:
-        logger.debug('Server closed with Keyboard Interrupt')
-    finally:
-        loop.close()
-
-
-def run_ipc_server(ipc_path, chain=None, loop=None):
-    server = start(ipc_path, chain, loop)
-    run_until_interrupt(server, loop)
-    if os.path.exists(ipc_path):
-        os.remove(ipc_path)
-
-
-if __name__ == '__main__':
-    logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
-    logging.getLogger('trinity.rpc.ipc').setLevel(logging.DEBUG)
-
-    run_ipc_server('/tmp/test.ipc')
+class IPCServer:
+    cancel_token = None
+    ipc_path = None
+    rpc = None
+    server = None
+
+    def __init__(self, rpc, ipc_path):
+        self.rpc = rpc
+        self.ipc_path = ipc_path
+        self.cancel_token = CancelToken('IPCServer')
+
+    async def run(self, loop=None):
+        self.server = await asyncio.start_unix_server(
+            connection_handler(self.rpc.execute, self.cancel_token),
+            self.ipc_path,
+            loop=loop,
```

Comment:
Although I usually prefer explicit over implicit, I've been a bit reluctant about ading the `loop` argument to our APIs as ISTM at this point it's only needed for tests, and we can work around that simply by having our fixtures depend on the `event_loop` fixture (even though we don't use the `event_loop`, like in https://github.com/ethereum/py-evm/blob/master/tests/p2p/test_kademlia.py#L22-L26). If you're ok with that I'd be happy to take care of documenting this properly and seeing if there's any way to provide better error messages when our tests fail because we end up accidentally using different loops